qbft/controller.go

```diff
@@ -33,10 +33,7 @@ func NewController(identifier []byte, committeeMember *types.CommitteeMember, co
 }
 
 // StartNewInstance will start a new QBFT instance, if can't will return error
-func (c *Controller) StartNewInstance(height Height, value []byte) error {
-	if err := c.GetConfig().GetValueCheckF()(value); err != nil {
-		return errors.Wrap(err, "value invalid")
-	}
+func (c *Controller) StartNewInstance(height Height, cdFetcher *types.DataFetcher) error {
```

Comment:
Why not move the fetching logic outside of controller and pass an already fetched value? this of course also means determining that we are leaders outside of the controller. IMO it makes more sense than passing a fetcher. less changes to controller.

Reply:
thinking again this might not work if we need to fetch data after round change? but probably we should fetch data at the beginning of the slot regardless.